### PR TITLE
Add MVP for Team 6

### PR DESCRIPTION
**Pull request by MattScottEEE**
# Summary

Added team6 imports to simulation.go and amended mapping in main.go (merge conflicts)

## Additional Information

Other than this, team6 agent implements social motives, messaging and food consumption 

## Test Plan

Tried out the driver code in main.go - it all works fine hehe :)

### CODE REVIEW

File: pkg/agents/team6/behaviours.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+	} else if newBehaviour < behaviourMin {
+		a.currBehaviour = behaviourMin
+	} else {
+		a.currBehaviour = newBehaviour
+	}
+
+}
+
+func (a *CustomAgent6) getBehaviourRange() (behaviourMax, behaviourMin behaviour) {
+
```
Comment (jzzheng22) [inline pkg/agents/team6/behaviours.go:49]:
```suggestion
```

File: pkg/agents/team6/behaviours.go [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+		a.currBehaviour = behaviourMax
+	} else if newBehaviour < behaviourMin {
+		a.currBehaviour = behaviourMin
+	} else {
+		a.currBehaviour = newBehaviour
+	}
+
+}
+
+func (a *CustomAgent6) getBehaviourRange() (behaviourMax, behaviourMin behaviour) {
```
Comment (jzzheng22) [inline pkg/agents/team6/behaviours.go:48]:
```suggestion
func (a *CustomAgent6) behaviourRange() (behaviourMax, behaviourMin behaviour) {
```

File: pkg/agents/team6/behaviours.go [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+	"math"
+	// "math/rand"
+)
+
+func (a *CustomAgent6) updateBehaviour() {
+
+	aConf := a.config
+	behaviourMax, behaviourMin := a.getBehaviourRange()
+
+	hpScore := 1 - float64(a.HP())/100.0 //map (minHP,maxHP) -> (1,0)
```
Comment (jzzheng22) [inline pkg/agents/team6/behaviours.go:13]:
```suggestion
	hpScore := 1 - float64(a.HP())/100.0
```

File: pkg/agents/team6/behaviours.go [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+	floor := a.Floor() + 1
+	// floor := rand.Intn(600) //* 600 //use for debug
```
Comment (jzzheng22) [inline pkg/agents/team6/behaviours.go:15]:
```suggestion
```

File: pkg/agents/team6/behaviours.go [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+
+	weights := aConf.paramWeights
+	behaviourPrediction := 0.0
+
+	for i := range weights {
+		behaviourPrediction += behaviourParams[i] * weights[i]
+	}
+
+	updateDir := behaviour(behaviourPrediction)*aConf.maxBehaviourThreshold - a.currBehaviour //find direction we need to move in to reach new behaviour prediction
+	updateMag := updateDir * behaviour(1-aConf.stubbornness)                                  //scale movement by stubbornness (minStubborn, maxStubborn) -> (fullMovement, 0)
```
Comment (jzzheng22) [inline pkg/agents/team6/behaviours.go:35]:
```suggestion
        // Find new direction required to reach new behaviour prediction
	updateDir := behaviour(behaviourPrediction)*aConf.maxBehaviourThreshold - a.currBehaviour
	// Scale movement by stubbornness (minStubborn, maxStubborn) -> (fullMovement, 0)
	updateMag := updateDir * behaviour(1-aConf.stubbornness)
```

File: pkg/agents/team6/behaviours.go [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+	newBehaviour := a.currBehaviour + updateMag
+
+	if newBehaviour > behaviourMax { //limit behaviour to max swing
+		a.currBehaviour = behaviourMax
+	} else if newBehaviour < behaviourMin {
+		a.currBehaviour = behaviourMin
+	} else {
+		a.currBehaviour = newBehaviour
+	}
+
```
Comment (jzzheng22) [inline pkg/agents/team6/behaviours.go:45]:
```suggestion
       	a.currBehaviour = behaviour(math.Max(float64(behaviourMin), math.Min(float64(newBehaviour), float64(behaviourMax))))
```

File: cmd/backend/main.go [hunk 7]
```diff
@@ -7,9 +7,9 @@ import (
 func main() {
 	// can have frontend parameters come go straight into simEnv
 	foodOnPlatform := 100.0
-	numOfAgents := []int{5, 5}
+	numOfAgents := []int{0,0,2}
```
[the remark this answers is not in the document]
Comment (jzzheng22) [reply cmd/backend/main.go:10]: Did you run gofmt on this??

File: pkg/agents/team6/behaviours.go (hunk 6 above)
Comment (jzzheng22) [reply pkg/agents/team6/behaviours.go:45]: Not actually sure if this suggestion is better... I'll think about it some more tomorrow

**Review by jzzheng22 — CHANGES_REQUESTED** [review]
You need to change your loggers to use the ones that were released earlier.

**Comment by jzzheng22** [comment]
Please make sure you've run `gofmt` on your files... if you're using VSCode this be done automatically when you save?

**Comment by jzzheng22** [comment]
And make sure your branch is up-to-date with no merge conflicts

File: pkg/agents/team6/config.go [hunk 8]
```diff
@@ -0,0 +1,113 @@
...
+				return v.bType
+			}
+		}
+	}
+
+	return fmt.Sprintf("UNKNOWN Behaviour '%v'", int(b))
+}
+
+func (a *CustomAgent6) Run() {
+	log.Printf("Custom agent %s", a.ID())
```
Comment (JaafarRammal) [inline pkg/agents/team6/config.go:93]: there is a new `a.Log()` function
- it has by default the ID and type of the agent printed
- on the top level we redirect where the logs go
- outputs in json format
- you can append fields to it
would be great if you can use the `a.Log()` instead 😄

**Review by JaafarRammal — CHANGES_REQUESTED** [review]
There is a new logging functionality to use instead of the `log.Printf`, would be great if you could migrate towards it

File: pkg/messages/team6Message.go [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+	msg := &Team6Message{
+		baseMessage: NewBaseMessage(SenderFloor),
+		Ack:         ack,
+	}
+	return msg
+}
+
+func (msg Team6Message) MessageType() string {
+	return "Wagwan babes"
+}
```
[the remark this answers is not in the document]
Comment (MattScottEEE) [reply pkg/messages/team6Message.go:18]: Placeholder name to test out if we were able to get this messaging system working

File: pkg/agents/team6/behaviours.go (hunk 6 above)
Comment (jzzheng22) [reply pkg/agents/team6/behaviours.go:44]: Okay I think the original is more readable, ignore the suggestion here.

File: pkg/messages/team6Message.go (hunk 9 above)
[the remark this answers is not in the document]
Comment (jzzheng22) [reply pkg/messages/team6Message.go:18]: Okay, but I'm not sure if this should be put into main

**Review by jzzheng22 — APPROVED** [review]

**Review by JaafarRammal — APPROVED** [review]
LGTM

**Pull request merged**